nautobot/core/ui/object_detail.py

```diff
@@ -378,25 +384,111 @@ def __init__(
         self,
         *,
         table_key,
+        max_display_count=None,
+        include_fields=None,
+        exclude_fields=None,
+        add_button_route="default",
+        add_permissions=None,
+        related_field_name=None,
         body_wrapper_template_path="components/panel/body_wrapper_table.html",
         body_content_template_path="components/panel/body_content_table.html",
+        header_extra_content_template_path="components/panel/header_table.html",
+        footer_content_template_path="components/panel/footer_table.html",
         **kwargs,
     ):
         """Instantiate an ObjectsTable panel.
 
         Args:
             table_key (str): The render context key string that contains the BaseTable instance of interest.
+            max_display_count (int, optional):  Maximum number of items to display in the table.
+                If None, defaults to the user's preference or a global setting.
```

Comment:
Can you clarify/elaborate on this? Maybe just reference `get_paginate_count()` if that's what makes the decision?